py/mpconfig.h

```diff
@@ -644,6 +644,16 @@ typedef double mp_float_t;
 #define MP_PLAT_FREE_EXEC(ptr, size) m_del(byte, ptr, size)
 #endif
 
+// Cast mp_obj_t to object pointer
+#ifndef MP_OBJ_CAST
```

Comment:
I don't think these should be here.  They should be in obj.h.  They are not for a port to configure, but rather for uPy itself to configure based on what object model is being used.